### PR TITLE
Add propertyname to gettransaction_MP call for SP create txs

### PR DESCRIPTION
**Pull request by zathras-crypto**

### CODE REVIEW

**Comment by ghost** [comment]
Looks like a good change to me

**Comment by dexX7** [comment]
I agree, most responses are pretty general, but miss some transaction type specific elements, so this is the right direction.

What confuses me, to be honest, is the fact that there is `gettransaction_MP`, `listtransactions_MP` and `searchtransactions_MP`, all sort of providing similar data, namely transaction information or an array of such transaction information, but this commit seems to affect only `gettransaction_MP`? _(edit: I noticed after looking at the code that it's already been there)_

Would it be possible to isolate and reuse the code in all three methods or does each one intends to provide different details?

As sidenote, because the context seems fitting and I'm wondering for some time now: what's the difference between `listtransactions_MP` and `searchtransactions_MP` besides that one is extended by an optional `address` field?

**Comment by zathras-crypto** [comment]
Hey DexX,

This PR was one of of need to be honest for my work integrating Core into MasterChest.info (if a create property tx is invalid, we don't add the details to the property map and so it's impossible to retrieve the property name for a valid:false property creation) - this just allows us to see the name regardless of validity.  Eventually we'll want to be able to retrieve all fields regardless of validity, but name is the key one I needed for my integration work.

We have an upcoming discussion where we're going to try and reign in the scope on RPC a little and make things a bit more standardized, including reusing code where possible, would welcome your input in those and will ping you when we're going to have them.

TL:DR; answer to your questions though:
gettransaction_MP is intended as the defacto call to get info about a transaction.
listtransactions_MP is intended to be a high level summary of recent transactions, where each transaction could then be interrogated further with gettransaction_MP if more info was required.
searchtransactions_MP is a slightly different call where the name implies that it could take some time to return a response due to the filter.

Thanks
Z

**Comment by dexX7** [comment]
Please do, I'm very interested.

Would you mind to quickly tell the difference between listtransactions_MP and searchtransactions_MP? Imho both do the roughly same and since listtransactions_MP does the same, but less than searchtransactions_MP, it might as well be removed?

**Comment by zathras-crypto** [comment]
Sure.  So basically, the issue is that we are looping the wallet's transactions looking for (by default) 10 transactions that match the address filter.  In the case that you have say 10,000 bitcoin txs and only a few (<10) MP transactions for the address, we will attempt to parse every one of those 10,000 bitcoin txs looking for our 10 MP transactions, making the call take a long time to return.  Thus the original call was renamed "searchtransactions_MP" and the filter removed from listtransactions_MP to minimize the chances of this happening.

FYI, I do intend to suggest we remove searchtransactions_MP soon btw and add the filter back into listtransactions_MP.  This is because recently I've changed that code to query levelDB for the existence of the tx before attempting to parse - we therefore no longer attempt to parse every non MP transaction in the wallet making the calls order of magnitudes faster - basically making the whole issue above moot.

Hope that makes sense!

**Comment by dexX7** [comment]
Ah I see, it actully does make sense and I understand the reasoning from your POV, but I'd add: this seems to be driven by low level implementation details and as user my intend is simply to "fetch transactions" which is likely not going to change, even if you switch from iterating wallet txs to much faster leveldb queries, nor do I really care about what is actually done to fullfil my intend. :)
